met-api/src/met_api/services/authorization.py

```diff
@@ -21,12 +21,13 @@ def check_auth(**kwargs):
     """Check if user is authorized to perform action on the service."""
     skip_tenant_check = current_app.config.get('IS_SINGLE_TENANT_ENVIRONMENT')
     user_from_context: UserContext = kwargs['user_context']
+    user_from_db = StaffUserModel.get_user_by_external_id(user_from_context.sub)
     token_roles = set(user_from_context.roles)
     permitted_roles = set(kwargs.get('one_of_roles', []))
     has_valid_roles = token_roles & permitted_roles
     if has_valid_roles:
         if not skip_tenant_check:
-            user_tenant_id = user_from_context.tenant_id
+            user_tenant_id = user_from_db.tenant_id
```

Comment:
If I understand correctly, you switched accessing the "current" user from app context to pulling that user from the DB. Was the former solution causing issues?